x/leverage/keeper/validate.go

```diff
@@ -9,6 +9,9 @@ import (
 // validateAcceptedDenom validates an sdk.Coin and ensures it is a registered Token
 // with Blacklisted == false
 func (k Keeper) validateAcceptedDenom(ctx sdk.Context, denom string) error {
+	if types.HasUTokenPrefix(denom) {
+		return types.ErrUToken.Wrap(denom)
+	}
 	token, err := k.GetTokenSettings(ctx, denom)
 	if err != nil {
```

Comment:
Adding this check before `GetTokenSettings`, which would fail anyway, changes the error from `not a registered denom: u/uumee` to `denom should not be a uToken: u/uumee`, which is more accurate.